client/my-sites/email/email-management/home/email-mailbox-warnings.jsx

```diff
@@ -0,0 +1,155 @@
+/**
+ * External dependencies
+ */
+import { Button } from '@automattic/components';
+import PropTypes from 'prop-types';
+import React from 'react';
+import { useDispatch } from 'react-redux';
+import { useTranslate } from 'i18n-calypso';
+
+/**
+ * Internal dependencies
+ */
+import {
+	EMAIL_WARNING_SLUG_GOOGLE_MAILBOX_TOS,
+	EMAIL_WARNING_SLUG_UNVERIFIED_FORWARDS,
+} from 'calypso/lib/emails/email-provider-constants';
+import {
+	getEmailForwardAddress,
+	hasGoogleAccountTOSWarning,
+	isGoogleEmailAccount,
+} from 'calypso/lib/emails';
+import { getGmailUrl } from 'calypso/lib/gsuite';
+import Gridicon from 'calypso/components/gridicon';
+import { isEmailForwardAccount } from 'calypso/lib/emails/is-email-forward-account';
+import { recordTracksEvent } from 'calypso/state/analytics/actions';
+import { resendVerificationEmail } from 'calypso/state/email-forwarding/actions';
+
+const EmailMailboxWarningText = ( { text } ) => {
+	return (
+		<div className="email-mailbox-warnings__warning">
+			<Gridicon icon="info-outline" size={ 18 } />
+			<span>{ text }</span>
+		</div>
+	);
+};
+
+const EmailMailboxWarningAction = ( { buttonText, isExternal, ...otherProps } ) => {
+	return (
+		<div className="email-mailbox-warnings__action">
+			<Button compact { ...otherProps }>
+				<span>{ buttonText }</span>
+				{ isExternal && <Gridicon icon="external" /> }
+			</Button>
+		</div>
+	);
+};
+
+const resendEmailForwardVerification = ( mailbox, dispatch ) => {
+	const destination = getEmailForwardAddress( mailbox );
+	dispatch(
+		recordTracksEvent(
+			'calypso_email_management_email_forwarding_resend_verification_email_click',
+			{
+				destination,
+				domain_name: mailbox.domain,
+				mailbox: mailbox.mailbox,
+			}
+		)
+	);
+	dispatch( resendVerificationEmail( mailbox.domain, mailbox.mailbox, destination ) );
+};
+
+const getDetailsForWarning = ( { account, dispatch, mailbox, translate, warning } ) => {
+	const warningSlug = warning.warning_slug;
+
+	if ( isGoogleEmailAccount( account ) ) {
+		// If we need the account-level ToS agreed to, just show a pending message and no CTA.
+		if ( hasGoogleAccountTOSWarning( account ) ) {
+			return {
+				warningText: translate( 'Pending' ),
+			};
+		}
+
+		if ( warningSlug === EMAIL_WARNING_SLUG_GOOGLE_MAILBOX_TOS ) {
+			const emailAddress = `${ mailbox.mailbox }@${ mailbox.domain }`;
```

Comment:
This may now need to be pulled into a `getEmailAddress` style selector as it been duplicated multiple times ...

Reply:
I'l take this on in a follow-up PR tomorrow. You're right that this is getting used in multiple places.